subprojects/core/src/main/java/org/gradle/internal/operations/DefaultBuildOperationExecutor.java

```diff
@@ -58,47 +55,68 @@ public DefaultBuildOperationExecutor(
         ParallelismConfiguration parallelismConfiguration,
         BuildOperationIdFactory buildOperationIdFactory
     ) {
-        super(listener, clock::getCurrentTime, buildOperationIdFactory);
+        this.runner = new DefaultBuildOperationRunner(
+            currentBuildOperationRef,
+            clock::getCurrentTime,
+            buildOperationIdFactory,
+            () -> new ListenerAdapter(listener, progressLoggerFactory, clock)
```

Comment:
❓ Why does this need to be created lazy?

Reply:
It's not about lazy. The `ListenerAdapter` is stateful and needs to be created for every execution.

Reply:
Understood. Not sure what I would pass here. Probably a Noop implementation?

Reply:
I added a constructor without this factory for a no-op case.